storybook/stories/Examples/Pie/CustomActiveShapePieChart.stories.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-shadow */
+import React, { useState } from 'react';
+import { Args } from '@storybook/react';
+import { Pie, PieChart, ResponsiveContainer, Sector } from '../../../../src';
+import { PieSectorDataItem } from '../../../../src/polar/Pie';
+
+export default {
+  component: Pie,
+};
+
+const data = [
+  { name: 'Group A', value: 400 },
+  { name: 'Group B', value: 300 },
+  { name: 'Group C', value: 300 },
+  { name: 'Group D', value: 200 },
+];
+
+export const CustomActiveShapePie = {
+  render: (args: Args) => {
+    const [activeIndex, setActiveIndex] = useState<number>(0);
+
+    const onPieEnter = (_: void, index: number) => {
+      setActiveIndex(index);
+    };
+
+    const renderActiveShape = (props: PieSectorDataItem) => {
+      const RADIAN = Math.PI / 180;
+      // @ts-expect-error property payload does not exist on type PieSectorDataItem
```

Comment:
Optional: Should we add it then? I like this PR because of its very clear scope and small size. 
That being said, type extensions specifically are non breaking, and I think are fine to include aside other changes.

Reply:
I'll do that in separate PR. One thing at a time

Reply:
https://github.com/recharts/recharts/pull/4030